18-websockets/classworks/Alexey_Goremykin/server.js

```diff
@@ -0,0 +1,17 @@
+const http = require('http');
+const server = http.createServer();
+const io = require('socket.io').listen(server);
+
+let messages =[];
+
+io.on('connection', socket => {
+  socket.on('send-message', (message) => {
+    messages.push(message)
+    io.sockets.emit('new-message', [...messages]);
+  });
+  socket.on('typing-message', (userName) => {
+    io.sockets.emit('typing-messasge-name', `Пользователь ${userName || 'Anonimus'} пишет сообщение`)
```

Comment:
I'd recommend sending to client only userName of typing user because you can have different locales. Also, let's send the event to all users instead of the current one.